packages/search/src/search.tsx

```diff
@@ -0,0 +1,417 @@
+/**
+ * External dependencies
+ */
+import React, { KeyboardEvent, MouseEvent } from 'react';
+import { debounce, noop, uniqueId } from 'lodash';
+import classNames from 'classnames';
+import { isMobile } from '@automattic/viewport';
+
+/**
+ * WordPress dependencies
+ */
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import { Spinner, __experimentalInputControl as InputControl } from '@wordpress/components';
```

Comment:
Originally I used `InputControl` because I didn't notice `TextControl`. However, I'm seeing that `TextControl` does not forward refs, making the logic dependent on the `searchInput` ref not work anymore. I'm testing to see if it's necessary in the first place.

Reply:
It turns out the forwarded ref is definitely necessary. I've opened a PR to add that to `TextControl`, though I think I would rather advocate for adopting the experimental `InputControl` which seems poised to eventually replace `TextControl` anyway.